content/cookbook/transactions/offline-transactions.md

```diff
@@ -0,0 +1,388 @@
+      1 * 10 ** tokenMint.decimals, // amount to transfer
+      tokenMint.decimals, // decimals of token
+    ),
+  );
+
+  // Partial sign as Bob
+  transaction.partialSign(bobKeypair);
+
+  // Serialize the transaction and convert to base64 to return it
+  const serializedTransaction = transaction.serialize({
+    // We will need Alice to deserialize and sign the transaction
+    requireAllSignatures: false,
+  });
+  const transactionBase64 = serializedTransaction.toString("base64");
+  return transactionBase64;
+
+  // The caller of this can convert it back to a transaction object:
+  const recoveredTransaction = Transaction.from(
+    Buffer.from(transactionBase64, "base64"),
+  );
+})();
+```
+
+## Durable Nonce
+
+`RecentBlockhash` is an important value for a transaction. Your transaction will
+be rejected if you use an expired recent blockhash (after 150 blocks). You can
+use `durable nonce` to get a never expired recent blockhash. To trigger this
+mechanism, your transaction must
+
+1. use a `nonce` stored in `nonce account` as a recent blockhash
+2. put `nonce advance` operation in the first instruction
+
+### Create Nonce Account
+
+```typescript filename="create-nonce-account.ts"
+import {
+  clusterApiUrl,
+  Connection,
+  Keypair,
+  Transaction,
+  NONCE_ACCOUNT_LENGTH,
+  SystemProgram,
+  LAMPORTS_PER_SOL,
+} from "@solana/web3.js";
+
+(async () => {
+  // Setup our connection and wallet
+  const connection = new Connection(clusterApiUrl("devnet"), "confirmed");
+  const feePayer = Keypair.generate();
+
+  // Fund our wallet with 1 SOL
+  const airdropSignature = await connection.requestAirdrop(
+    feePayer.publicKey,
+    LAMPORTS_PER_SOL,
+  );
+  await connection.confirmTransaction(airdropSignature);
+
+  const nonceAccountAuth = Keypair.generate();
+
+  let nonceAccount = Keypair.generate();
+  console.log(`nonce account: ${nonceAccount.publicKey.toBase58()}`);
+
+  let tx = new Transaction().add(
+    // create nonce account
+    SystemProgram.createAccount({
+      fromPubkey: feePayer.publicKey,
+      newAccountPubkey: nonceAccount.publicKey,
+      lamports:
+        await connection.getMinimumBalanceForRentExemption(
+          NONCE_ACCOUNT_LENGTH,
+        ),
+      space: NONCE_ACCOUNT_LENGTH,
+      programId: SystemProgram.programId,
+    }),
+    // init nonce account
+    SystemProgram.nonceInitialize({
+      noncePubkey: nonceAccount.publicKey, // nonce account pubkey
+      authorizedPubkey: nonceAccountAuth.publicKey, // nonce account authority (for advance and close)
+    }),
+  );
+
+  console.log(
+    `txhash: ${await sendAndConfirmTransaction(connection, tx, [feePayer, nonceAccount])}`,
+  );
+})();
+```
+
+### Get Nonce Account
+
+```typescript filename="get-nonce-account.ts"
+import {
+  clusterApiUrl,
+  Connection,
+  PublicKey,
+  Keypair,
+  NonceAccount,
+} from "@solana/web3.js";
+
+(async () => {
+  // connection
+  const connection = new Connection(clusterApiUrl("devnet"), "confirmed");
```

Comment:
This is up to you, but I think we can do without comments like this where the code is very obvious.